desktop/UpdateManager.ts

```diff
@@ -0,0 +1,62 @@
+import { BrowserWindow, dialog } from 'electron';
+import { autoUpdater } from 'electron-updater';
+import log from 'electron-log';
+import StoreService from './storeService';
+import { MessageBoxOptions } from 'electron/main';
+import NotificationManager from './notificationManager';
+export default class UpdateManager {
+  private mainWindow: BrowserWindow;
+  private notification: NotificationManager;
+
+  UPDATE_INTERVAL = 1000 * 60 * 60 * 24;
+
+  constructor(mainWindow: BrowserWindow) {
+    this.mainWindow = mainWindow;
+    this.notification = new NotificationManager(this.mainWindow);
+    if (process.env.NODE_ENV === 'production' && StoreService.get('userSettings.promptForUpdate')) {
+      log.transports.file.level = 'info';
+      autoUpdater.logger = log;
+
+      autoUpdater.channel = 'latest';
+      autoUpdater.allowDowngrade = false;
+
+      this.checkForUpdates();
+
+      autoUpdater.on('update-downloaded', this.onUpdateAvailable);
+
+      setInterval(() => autoUpdater.checkForUpdates(), this.UPDATE_INTERVAL);
+    }
+  }
+
+  async updateApplication() {
+    autoUpdater.downloadUpdate();
```

Comment:
Are you sure it will update App ASAP?

Reply:
Yes

Reply:
This works. I had to mock a dev update, but without authentication token, it fails.
```Error: Error: Cannot find latest-mac.yml in the latest release artifacts (https://github.com/spacemeshos/smapp/releases/download/v0.1.21-ntp-fix/latest-mac.yml): HttpError: 404 
"method: GET url: https://github.com/spacemeshos/smapp/releases/download/v0.1.21-ntp-fix/latest-mac.yml\n\nPlease double check that your authentication token is correct. Due to security reasons actual status maybe not reported, but 404.\n"
Headers: {
  "server": "GitHub.com",
  "date": "Mon, 13 Sep 2021 11:48:31 GMT",
  "content-type": "text/plain; charset=utf-8",
  "vary": "X-PJAX, X-PJAX-Container, Accept-Encoding, Accept, X-Requested-With",
  "permissions-policy": "interest-cohort=()",
  "cache-control": "no-cache",
  "strict-transport-security": "max-age=31536000; includeSubdomains; preload",
  "x-frame-options": "deny",
  "x-content-type-options": "nosniff",
  "x-xss-protection": "0",
  "referrer-policy": "no-referrer-when-downgrade",
  "expect-ct": "max-age=2592000, report-uri=\"https://api.github.com/_private/browser/errors\"",
  "content-security-policy": "default-src 'none'; base-uri 'self'; connect-src 'self'; form-action 'self'; img-src 'self' data:; script-src 'self'; style-src 'unsafe-inline'",
  "content-encoding": "gzip",
  "content-length": "29",
  "x-github-request-id": "F559:85D2:22B4052:2436836:613F3A8E"
}
    at createHttpError ...```

Reply:
Yep, it tried to download the update.
However, I think that the App won't be updated, because to update the App we have to call `quitAndInstall` method on autoUpdater (https://www.electron.build/auto-update.html)
Did I miss something?

Reply:
Replacing this with `autoUpdater.quitAndInstall()` because the event is `update-downloaded` and not `update-available`. Thanks for catching the bug.